### DISCUSSION TITLE
How do I get the size of the current Widget/Its parent?

### DISCUSSION THREAD

**Discussion by ciaransweet** [Textualize/textual#642] [Q&A]
I have a scenario like so:

```python
class MyApp(App)
    ...
    async def on_mount(self) -> None:
        await self.view.dock(MyWidget())
        
class MyWidget(Widget):
     def __init__(self):
         self._size = # my/or my parents size?

MyApp.run()
```

I'd like to know how I can determine the current widgets size and/or the size of the dock I guess?

I can see the Resize event sizes, but I've tried many combinations of `self.size`, `view.size` etc and everything gives me back `0`'s across the board 😭

**Comment by willmcgugan** [comment]
You can't do it on init. It won't have a size until the layout process which happens on refresh. It should be available on the Widget's Mount event.

**Reply by ciaransweet** [reply]
Ah okay, do you have an example of that event being processed anywhere?

```python
     def on_event(self, event: events.Event) -> None:
        if event == events.Mount:
            textual.log(self.size)
```

Seems to just lock up the app if I implement it on the widget
